### ISSUE TITLE
powershell hangs from ssh session

### ISSUE THREAD

**Issue by covici** [PowerShell/Win32-OpenSSH#188]
I was using ssh on Linux into my windows 10 box using sshd from this repository and when I ran powershell, it hung and I had to kill the ssh Linux client and could not reconnect till I restarted the sshd service.

**Comment by scottmckenzie** [comment]
Are you using the syntax `powershell -File -` as per https://github.com/PowerShell/Win32-OpenSSH/wiki/ssh.exe-examples ?

**Comment by covici** [comment]
No, I want an interactive session with powershell.

Scott McKenzie notifications@github.com wrote:

> Are you using the syntax `powershell -File -` as per https://github.com/PowerShell/Win32-OpenSSH/wiki/ssh.exe-examples ?
> 
> ---
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/PowerShell/Win32-OpenSSH/issues/188#issuecomment-212157121
> 
> ---
> 
> Alternatives:
> 
> ---

## 

Your life is like a penny.  You're going to lose it.  The question is:
How do
you spend it?

```
     John Covici
     covici@ccs.covici.com
```

**Comment by scottmckenzie** [comment]
That is the syntax to get an interactive session with powershell.

**Comment by covici** [comment]
OK, you are correct, in a sense, but for some reason, once I execute
that command the  environment variables seem not to work.  I have  a
file called $HOME/something, but ps can't find it because it thinks home
is hnot set.  Now I notice that exiting out of the command prompt, I can
no longer open a new session  -- the connection just hangs.

Scott McKenzie notifications@github.com wrote:

> That is the syntax to get an interactive session with powershell.
> 
> ---
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/PowerShell/Win32-OpenSSH/issues/188#issuecomment-212253387
> 
> ---
> 
> Alternatives:
> 
> ---

## 

Your life is like a penny.  You're going to lose it.  The question is:
How do
you spend it?

```
     John Covici
     covici@ccs.covici.com
```

**Comment by scottmckenzie** [comment]
Are you using $env:HOME to access the environment variable?  This works for me in a powershell session.

**Comment by covici** [comment]
No, its just $home as part of a filename which imports my command
history.

Scott McKenzie notifications@github.com wrote:

> Are you using $env:HOME to access the environment variable?  This works for me in a powershell session.
> 
> ---
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/PowerShell/Win32-OpenSSH/issues/188#issuecomment-212377247
> 
> ---
> 
> Alternatives:
> 
> ---

## 

Your life is like a penny.  You're going to lose it.  The question is:
How do
you spend it?

```
     John Covici
     covici@ccs.covici.com
```

**Comment by scottmckenzie** [comment]
`$home` is a powershell variable.  If you have not assigned it a value it will be null.  To access environment variables in powershell you must you the syntax `$Env:ENVVARIABLENAME`.
https://technet.microsoft.com/en-us/library/ff730964.aspx

**Comment by covici** [comment]
hmmm, on my local session, I don't have this problem i.e. in my user
variables I have a value for HOME which is recognized by powershell.
But I guess when logging in through ssh, those user variables are not
recognized by powershell when I try to run it from ssh.

Scott McKenzie notifications@github.com wrote:

> `$home` is a powershell variable.  If you have not assigned it a value it will be null.  To access environment variables in powershell you must you the syntax `$Env:ENVVARIABLENAME`.
> https://technet.microsoft.com/en-us/library/ff730964.aspx
> 
> ---
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/PowerShell/Win32-OpenSSH/issues/188#issuecomment-213151562
> 
> ---
> 
> Alternatives:
> 
> ---

## 

Your life is like a penny.  You're going to lose it.  The question is:
How do
you spend it?

```
     John Covici
     covici@ccs.covici.com
```

**Comment by joeyaiello** [comment]
@covici is right here, `$HOME` is an [automatic variable](https://technet.microsoft.com/en-us/library/hh847768.aspx) that should be populated regardless of whether you're using SSH. 

I'm not getting a repro here. I logged on using password-based auth (both on a domain-joined account and a local account), and `$HOME` works fine. Didn't try with key-based auth though, are you using that? 

There is an extra newline above the path itself when it's get spit out to the console. Maybe that's what's giving you grief? Although as long as you're using the *-Path cmdlets you should be okay (`$HOME | Test-Path` returns `True` for me).

**Comment by covici** [comment]
I am using password authentication.  If I log in and do a set  command
with no arguments, I  do see home, but powershell does not, it seems.
At least, it is not seeing it initially, once in there if I do
set-location env:, HOME does show up, if I do dir, so I am not sure what
is happening here.
I wish tab would work, otherwise powershell is  pretty hard to use.

Joey Aiello notifications@github.com wrote:

> @covici is right here, `$HOME` is an [automatic variable](https://technet.microsoft.com/en-us/library/hh847768.aspx) that should be populated regardless of whether you're using SSH. 
> 
> I'm not getting a repro here. I logged on using password-based auth (both on a domain-joined account and a local account), and `$HOME` works fine. Didn't try with key-based auth though, are you using that? 
> 
> There is an extra newline above the path itself when it's get spit out to the console. Maybe that's what's giving you grief? Although as long as you're using the *-Path cmdlets you should be okay (`$HOME | Test-Path` returns `True` for me). 
> 
> ---
> 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/PowerShell/Win32-OpenSSH/issues/188#issuecomment-213579551
> 
> ---
> 
> Alternatives:
> 
> ---

## 

Your life is like a penny.  You're going to lose it.  The question is:
How do
you spend it?

```
     John Covici
     covici@ccs.covici.com
```

**Comment by covici** [comment]
Well, something strange is happening, when I invoke ps from the host,  I
see lots more environment variables  maybe because I am running as
administrator -- so how can I do that from the ssh?

Joey Aiello notifications@github.com wrote:

> @covici is right here, `$HOME` is an [automatic variable](https://technet.microsoft.com/en-us/library/hh847768.aspx) that should be populated regardless of whether you're using SSH. 
> 
> I'm not getting a repro here. I logged on using password-based auth (both on a domain-joined account and a local account), and `$HOME` works fine. Didn't try with key-based auth though, are you using that? 
> 
> There is an extra newline above the path itself when it's get spit out to the console. Maybe that's what's giving you grief? Although as long as you're using the *-Path cmdlets you should be okay (`$HOME | Test-Path` returns `True` for me). 
> 
> ---
> 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/PowerShell/Win32-OpenSSH/issues/188#issuecomment-213579551
> 
> ---
> 
> Alternatives:
> 
> ---

## 

Your life is like a penny.  You're going to lose it.  The question is:
How do
you spend it?

```
     John Covici
     covici@ccs.covici.com
```

**Comment by discodance1** [comment]
I am seeing the same issue interactive session with powershell, when ssh'ing from my linux server to a windows server.
I try to connect to a powershell constrained end point from linux:
"Enter-pssession -ConfigurationName <name> "

next instead of asking for computername as in normal powershell console. openssh+ powerhsell just hangs and i can't even exit out of that console.

**Comment by discodance1** [comment]
Is there an update on this ?I have tried with the latest version of win32-openssh 64 bit on my windows server 2012 R2 standard. 

When i am able to ssh from my linux server to windows server but when i run 

> powershell -File -

I get an error saying" The shell cannot be started. A failure occured during initialization: The path is not of a legal form."

**Comment by troshko111** [comment]
Ran into this today - starting `PowerShell` completely kills ssh server (2012R2 DC) and client. Any ETA for this one? It actually sounds pretty serious blocking issue

**Comment by DarwinJS** [comment]
FYI - tested this on Nano RTM and it works fine - powershell launches and CMDLets are not slow - I think because it shipped with PSH 5.1 (but .NET core, not 4.6)

**Comment by manojampalam** [comment]
Try out the latest release. All native Windows console apps (including Powershell.exe) should work now over remote ssh sessions.
